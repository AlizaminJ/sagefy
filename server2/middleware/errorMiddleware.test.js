const errorMiddleware = require('./errorMiddleware')
const abort = require('../helpers/abort')

function createRes() {
  return {
    status(code) {
      this.xcode = code
      return this
    },
    json(o) {
      this.xjson = o
      return this
    },
  }
}

describe('#errorMiddleware', () => {
  test('handle a status code error', () => {
    const err = abort(401)
    const req = {}
    const res = createRes()
    const next = jest.fn()
    errorMiddleware(err, req, res, next)
    expect(res).toMatchSnapshot()
    expect(next).toMatchSnapshot()
  })

  test('handle a joi validation error', () => {
    const err = {
      isJoi: true,
      details: {
        message: '"name" is not allowed to be empty',
        path: ['name'],
        type: 'any.empty',
      },
    }
    const req = {}
    const res = createRes()
    const next = jest.fn()
    errorMiddleware(err, req, res, next)
    expect(res).toMatchSnapshot()
    expect(next).toMatchSnapshot()
  })

  test('handle a 500 error', () => {
    const err = new Error()
    const req = {}
    const res = createRes()
    const next = jest.fn()
    errorMiddleware(err, req, res, next)
    expect(res).toMatchSnapshot()
    expect(next).toMatchSnapshot()
  })

  test('pass through', () => {
    const err = null
    const req = {}
    const res = createRes()
    const next = jest.fn()
    errorMiddleware(err, req, res, next)
    expect(res).toMatchSnapshot()
    expect(next).toMatchSnapshot()
  })
})
