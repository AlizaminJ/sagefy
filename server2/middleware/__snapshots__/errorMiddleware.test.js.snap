// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#errorMiddleware handle a 500 error 1`] = `
Object {
  "json": [Function],
  "status": [Function],
  "xcode": 500,
  "xjson": Object {
    "errors": Array [
      Object {
        "message": "Server error.",
      },
    ],
    "ref": "UilnKcaD20CQRWAFc4s8Eg",
  },
}
`;

exports[`#errorMiddleware handle a 500 error 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Error],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`#errorMiddleware handle a joi validation error 1`] = `
Object {
  "json": [Function],
  "status": [Function],
  "xcode": 400,
  "xjson": Object {
    "errors": Array [
      Object {
        "message": "\\"name\\" is not allowed to be empty",
        "name": "name",
        "type": "any.empty",
      },
    ],
    "ref": "2lEH29U46EukhyCL_KlgMQ",
  },
}
`;

exports[`#errorMiddleware handle a joi validation error 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "details": Object {
          "message": "\\"name\\" is not allowed to be empty",
          "path": Array [
            "name",
          ],
          "type": "any.empty",
        },
        "isJoi": true,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`#errorMiddleware handle a status code error 1`] = `
Object {
  "json": [Function],
  "status": [Function],
  "xcode": 401,
  "xjson": Object {
    "errors": Array [
      Object {
        "message": "Unauthorized",
      },
    ],
    "ref": undefined,
  },
}
`;

exports[`#errorMiddleware handle a status code error 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [Error],
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`#errorMiddleware pass through 1`] = `
Object {
  "json": [Function],
  "status": [Function],
}
`;

exports[`#errorMiddleware pass through 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      null,
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
